docs/federate.md

```diff
@@ -0,0 +1,135 @@
+Setting up Federation
+=====================
+
+Federation is the process by which users on different servers can participate
+in the same room. For this to work, those other servers must be able to contact
+yours to send messages.
+
+The ``server_name`` configured in the Synapse configuration file (often
+``homeserver.yaml``) defines how resources (users, rooms,...) will be
+identified (ie: ``@user:example.com``, ``#room:example.com``). By
+default, it is also the domain that other servers will use to
+try to reach your server via port 8448. This is easy to set
+up and will work provided you set the ``server_name`` to match your
+machine's public DNS hostname, and provide Synapse with a TLS certificate
+which is valid for your ``server_name``.
+
+Once you have completed the steps necessary to federate, you should be able to
```

Comment:
given that you should be able to connect with a client before this, it feels slightly confusing to reiterate this - I am left wondering if I should now be using a different method to connect. In short, I'd be inclined to miss out the "you should be able to connect" phrase and go straight for "you should be able to join rooms via federation".

Reply:
Agreed